### ISSUE TITLE
Search box is not cleared on Microsoft Surface

### ISSUE THREAD

**Issue by ko-vasilev** [ko-vasilev/Hearthstone-Collection-Tracker#25]
While running import on Surface search box is not being cleared resulting 0 cards in collection.
For testing contact valuequest.

**Comment by quark-zju** [comment]
I encountered the same bug. I thought this is because the UI change (searchbox moved and not cleared when clicked) probably not a Surface specified bug.

**Comment by ko-vasilev** [comment]
So it only inserts card names but does not clear search box right?

**Comment by quark-zju** [comment]
Yes. I think sending Ctrl+A before typing the card name will probably fix it.

**Comment by ko-vasilev** [comment]
I believe when you click on search box, it selects all text by default.

**Comment by quark-zju** [comment]
On my windows 10 surface pro, it won't select automatically on click.

**Comment by ko-vasilev** [comment]
Oh so you also has a surface.. If you're saying click does not select all the text, this might be the source of bug (because on my desktop Win7 it selects all text by default).
Thank you for the info!

**Comment by quark-zju** [comment]
Actually I suspect it's because Windows 10, not Surface.

HDT itself also has a similar problem when exporting Deck to Hearthstone because it uses the same technique. I remember clearly the exporting feature worked pretty well on my Surface on a previous Hearthstone version and Win 8. It seems the exporting stops working after some software upgrade (Win 10 or Hearthstone). Since Hearthstone works as before on your Win 7, I strongly suspect it's Windows 10. I will change HDT code to test it later today.

**Comment by ko-vasilev** [comment]
Oh okay, it totally might be Win10 fault.

**Comment by quark-zju** [comment]
I am experimenting some changes on HDT. However it seems it's not that trivial by just inserting a Ctrl+A. I will try to a reliable fix.

**Comment by quark-zju** [comment]
In my HS 3.0.0.9786, its search textbox has two locations, the red and green (they will not be both visible at the same time):

![HS Textboxe Locations](https://cloud.githubusercontent.com/assets/445459/9493116/f93dcadc-4c2f-11e5-91f6-1e80a827d396.png)

I will use R and G for short. And here is the rules I found in my computer, which are really confusing but can be reproduced:
- Click G toggles R and G. One becomes visible and the other becomes invisible.
- When R is visible, ENTER will filter the cards and G will become empty, normally (see below for the "abnormal" case).
- When R is visible, click G will filter the cards, normally.
- When G is visible, ENTER does NOT filter the cards, which is the main cause of the abnormal cases
- When G is visible and it is empty and the cards are already filtered by some search words, ENTER makes the X button disappear, making it impossible to reset the search using mouse.
- Pressing the red X on the right of the text box in either R or G will reset the search.

To reproduce the abnormal cases:
1. Click at G. R appears.
2. Type something, such as "north", DO NOT PRESS ENTER.
3. Click at G again. R disappears. G appears with "nor" selected. Cards are filtered.
4. Type something, such as "nova", DO NOT PRESS ENTER.
5. Click at G again. G disappears. R appears with "nova" and the cursor is at the right of "a", the text is NOT selected.
6. Press ENTER or click at G. G appears with text "nova" selected and R disappears. The cards should but are NOT filtered by the word "nova".

Another one, which is probably the reason why HDT exporting does not works properly:
1. Click at G. Type "north" in R. Press ENTER.
2. Press the red cross button X in G. 
3. Click at G. R should be focused but it is not, G is focused instead. Type anything and press ENTER won't filter cards as expected.

It looks like a Blizzard bug. I will open link this comment to a HDT issue to discuss it later. While I am still not fully understand why HDT exporting won't work. It seems pressing "ENTER" twice does the trick.

**Comment by ko-vasilev** [comment]
Uh... Maybe it is surface-only or Win10 only? I never saw that red search field..

**Comment by quark-zju** [comment]
WIn 7 compatibility mode can workaround the issue while Win 8 can't.

**Comment by tgfish** [comment]
I just want to chime in to say that I also was unable to import my collection on my Surface Pro 4 running Windows 10. It's definitely related somehow to that search box jumping around and something odd with having to hit enter twice (I already noticed this annoyance when manually searching). Running Hearthstone in Windows 7 Compatibility mode resolved the issue.
